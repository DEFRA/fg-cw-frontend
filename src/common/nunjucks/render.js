import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
