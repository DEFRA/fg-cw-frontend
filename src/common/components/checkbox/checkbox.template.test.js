import { describe, expect, test } from "vitest";
import { render } from "../../nunjucks/render.js";

describe("checkbox", () => {
  test("renders", () => {
    const component = render("checkbox", {
      id: "checkbox-id",
      name: "checkbox-name",
      label: "Checkbox Label",
      value: "checkbox-value",
      suffix: "Checkbox Suffix",
    });

    expect(component).toMatchSnapshot();
  });
});
