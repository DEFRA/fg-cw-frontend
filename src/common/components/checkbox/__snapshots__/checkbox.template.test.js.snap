// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checkbox > renders 1`] = `
"<div
  class="govuk-checkboxes__item govuk-checkboxes--small moj-multi-select__checkbox"
>
  <input
    type="checkbox"
    class="govuk-checkboxes__input"
    id="checkbox-idCheckbox Suffix"
    name="checkbox-name"
    value="checkbox-value"
  />
  <label
    class="govuk-label govuk-checkboxes__label"
    for="checkbox-idCheckbox Suffix"
  >
    <span class="govuk-visually-hidden">Select Checkbox Label</span>
  </label>
</div>
"
`;
